src/Components/Body/Body.tsx

```diff
@@ -1,97 +1,30 @@
-  user: string | null;
-}
-interface userSubscriptions {
-  id: string;
-  name: string;
-  directMessages: string[];
-  channels: string[];
-}
-type chatInformation = {
-  id: string;
-  name: string;
-  userID: string[];
-  messages: string[];
-  type: string;
-};
-function Body({ user }: bodyProps) {
-  const { data, loading } = useQuery<userSubscriptions>({
-    url: `user/id?userId=${user}`,
-    method: "GET",
-  });
+import { ChatArea } from "./ChatArea/ChatArea";
+import { NewChannelForm } from "./NewChannelForm/NewChannelForm";
+import { useModal } from "../../Hooks/useModal";
+import "./Body.css";
 
-  const [selectedChat, setSelectedChat] = useState("");
+function Body() {
   const [selectedChatId, setSelectedChatId] = useState("");
-  const [messages, setMessages] = useState<string[]>([]);
-  const [newMessage, setNewMessage] = useState<string>("");
-  const { mutate } = useMutation<chatInformation>((data) => {
-    console.log(newMessage, "before post call");
-    const url = `${ROOT_URL}chat/id`;
-    console.log(`${ROOT_URL}chat/id`);
-    const options = {
-      method: "POST",
-      body: JSON.stringify(data),
-      headers: {
-        "Content-Type": "application/json",
-      },
-    };
-    return fetch(url, options);
-  });
-
-  const handleSelectedChat = (activeChat: string): void => {
-    setSelectedChat(activeChat);
-  };
+  const {
+    isModalOpen: isNewChannelFormOpen,
+    toggleModalState: toggleNewChannelForm,
+  } = useModal();
 
-  const handleSelectedChatId = (activeChatId: string): void => {
+  const handleSelectedChatId = useCallback((activeChatId: string): void => {
     setSelectedChatId(activeChatId);
-  };
-
-  const handleChatMessages = (curMessages: string[]): void => {
-    setMessages(curMessages);
-  };
-
-  const addNewMessage = (newMessageId: string) => {
-    const payload = {
-      chatId: selectedChatId,
-      messageId: newMessageId,
-    };
-    mutate(payload);
-    setMessages((prevMessages) => {
-      console.log([...prevMessages, newMessage],newMessage, "inside setMessages");
-      return [...prevMessages, newMessage];
-    });
-    setNewMessage(newMessageId);
-    console.log(messages, "added new message");
-  };
+  }, []);
 
   return (
     <div className="Body-container">
-      {!loading && (
-        <>
-          <Sidepanel
-            channels={data ? data.channels : []}
-            friends={data ? data.directMessages : []}
-            handleSelectedChat={handleSelectedChat}
-            handleSelectedChatId={handleSelectedChatId}
-            handleChatMessages={handleChatMessages}
-          />
-          <MessageContainer
-            sender={user}
-            activeChat={selectedChat}
-            activeChatId={selectedChatId}
-            messages={messages}
-            addNewMessage={addNewMessage}
-          />
-        </>
-      )}
+      <Sidepanel
+        selectedListItem={selectedChatId}
+        handleSelectedListItem={handleSelectedChatId}
```

Comment:
follow the following naming convention for handlers.

prop name should be `onEventName` and the prop passed should be `handleEventName`

e.g. onClick, onDrag, onSubmit etc.